istioctl/cmd/sidecar.go

```diff
@@ -0,0 +1,361 @@
+	iSpec, err := unstructureIstioType(spec)
+	if err != nil {
+		return nil, err
+	}
+	u.Object["spec"] = iSpec
+	// remove creationTimestamp left behind by ObjectMeta
+	delete(iSpec["template"].(map[string]interface{})["metadata"].(map[string]interface{}), "creationTimestamp")
+
+	wgYAML, err := yaml.Marshal(u.Object)
+	if err != nil {
+		return nil, err
+	}
+	return wgYAML, nil
+}
+
+// Cluster inference from the current kubectl context only works for GKE
+func generateConfigCommand() *cobra.Command {
+	generateConfigCmd := &cobra.Command{
+		Use:   "generate-config",
+		Short: "Generates and packs all the required configuration files for deployment",
+		Long: `Takes in WorkloadGroup artifact, then generates and packs all the required configuration files for deployment. 
+This includes a MeshConfig resource, the cluster.env file, and necessary certificates and security tokens. 
+Tries to automatically infer the target cluster, and prompts for flags if the cluster cannot be inferred`,
+		Example: "generate-config -f workloadgroup.yaml -o config",
+		Args: func(cmd *cobra.Command, args []string) error {
+			if filename == "" {
+				return fmt.Errorf("expecting a WorkloadGroup artifact file")
+			}
+			if outputName == "" {
+				return fmt.Errorf("expecting an output filename")
+			}
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			wg := &clientv1alpha3.WorkloadGroup{}
+			if err := readWorkloadGroup(filename, wg); err != nil {
+				return err
+			}
+			kubeClient, err := kube.NewExtendedClient(kube.BuildClientCmd(kubeconfig, configContext), "")
+			if err != nil {
+				return err
+			}
+
+			if err = createConfig(kubeClient, wg, clusterID, revision, outputName); err != nil {
+				return err
+			}
+			return nil
+		},
+	}
+	generateConfigCmd.PersistentFlags().StringVarP(&filename, "file", "f", "", "filename of the WorkloadGroup artifact")
+	generateConfigCmd.PersistentFlags().StringVarP(&outputName, "output", "o", "", "Name of the tarball to be created")
+
+	generateConfigCmd.PersistentFlags().StringVar(&clusterID, "cluster id", "", "The ID used to identify the cluster")
+	generateConfigCmd.PersistentFlags().StringVar(&revision, "revision", "", "control plane revision (experimental)")
+	return generateConfigCmd
+}
+
+func readWorkloadGroup(filename string, wg *clientv1alpha3.WorkloadGroup) error {
+	f, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return err
+	}
+	if err = yaml.Unmarshal(f, wg); err != nil {
+		return err
+	}
+	return nil
+}
+
+// Creates all the relevant config for the given workload group and cluster
+func createConfig(kubeClient kube.ExtendedClient, wg *clientv1alpha3.WorkloadGroup, clusterID, revision, outputName string) error {
+	temp, err := ioutil.TempDir("./", outputName)
+	if err != nil {
+		return err
+	}
+	defer os.RemoveAll(temp)
+
+	if err = createClusterEnv(wg, temp); err != nil {
+		return err
+	}
+	if err = createCertificates(kubeClient, temp); err != nil {
```

Comment:
certificates - or tokens - for istio 1.8? If tokens, the token procedure must be documented - I am not clear it is at the moment.